AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/helper/AndroidManifestFinder.java

```diff
@@ -193,22 +221,26 @@ boolean applies() {
 
 		private final AndroidAnnotationsEnvironment environment;
 
-		GradleAndroidManifestFinderStrategy(AndroidAnnotationsEnvironment environment, String sourceFolder) {
-			super("Gradle", GRADLE_GEN_FOLDER, sourceFolder);
+		AbstractGradleAndroidManifestFinderStrategy(Pattern pattern, AndroidAnnotationsEnvironment environment, String sourceFolder) {
+			super("Gradle", pattern, sourceFolder);
 			this.environment = environment;
 		}
 
+		protected String getPath() {
+			return matcher.group(1);
+		}
+
+		protected abstract String getGradleVariant();
+
 		@Override
 		Iterable<String> possibleLocations() {
-			String path = matcher.group(1);
-			String mode = matcher.group(2);
-			String gradleVariant = matcher.group(3);
-			String variantPart = gradleVariant.substring(1);
+			String path = getPath();
+			String gradleVariant = getGradleVariant();
 
 			List<String> possibleLocations = new ArrayList<>();
-			findPossibleLocationsV32(path, variantPart, possibleLocations);
+			findPossibleLocationsV32(path, gradleVariant, possibleLocations);
```

Comment:
For which reason the change in the name of the variable `variantPart`?

Reply:
Before this change we had `gradleVariant ` and `variantPart`. `variantPart` was a substring of `gradleVariant` and as this substring part is gone i decided to use `gradleVariant ` as the remaining variable.